### ISSUE TITLE
Remove tags before heading 3 above job titles

### ISSUE THREAD

**Issue by gobyrne** [GCTC-NTGC/gc-digital-talent#6952]
### Description

On browse jobs page. The classification appears before the job title in the reading order. This leads to confusing reading from the screen reader.

![Image](https://user-images.githubusercontent.com/2494551/245827473-880ba2a9-7016-43c8-94b6-6794c610cf4e.png)

> Remove the tag s that read before the Heading 3 titles for each job title. They don't make any sense read before the job title for screen readers. ."WCAG criteria: 1.3.1 Info and Relationships, 1.3.2 Meaningful Sequence
> 
> They say things like "ai-01," abbreviations can sometimes be confusing or misinterpreted unless they are explained). A link to a help section with the means would improve the accessibility for those using screen readers as sometimes screen readers like VoiceOver will try to read the letters as actual wordsWhen tabbing through the page, sometimes odd behaviour would occur on the links. Something about a missing abbreviation. Apply to this recruitment (Abbreviation not found.) Currently, the apply link only tells you what section the job opportunity is in. It would be good to repeat the job title as well.
> 

### Acceptance Criteria
- [ ] **Aria-hidden = true** on the recruitment flag (that shows the classification - circled in red in image)
